ddtrace/tracer/metrics.go

```diff
@@ -101,3 +118,25 @@ func (t *tracer) reportMetrics(statsd gauger, interval time.Duration) {
 		}
 	}
 }
+
+func (t *tracer) sendHealthMetrics() {
+	t.config.statsd.Count("datadog.tracer.spans_started", atomic.SwapInt64(&t.spansStarted, 0), nil, 1)
+	t.config.statsd.Count("datadog.tracer.spans_finished", atomic.SwapInt64(&t.spansFinished, 0), nil, 1)
+	t.config.statsd.Count("datadog.tracer.traces_dropped", atomic.SwapInt64(&t.tracesDropped, 0), []string{"reason:trace_too_large"}, 1)
+}
+
+func (t *tracer) reportHealthMetrics() {
+	defer t.wg.Done()
+	ticker := time.NewTicker(statsInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			t.sendHealthMetrics()
+		case <-t.stopped:
+			t.sendHealthMetrics()
+			t.config.statsd.Close()
```

Comment:
This seems like an odd place to close. Doesn't it make more sense to do it in the `worker()`? All the shutdown happens there, as a receive from the exit channel.